src/Microsoft.AspNet.Mvc.Xml/Formatters/XmlDataContractSerializerOutputFormatter.cs

```diff
@@ -45,30 +67,49 @@ public XmlDataContractSerializerOutputFormatter([NotNull] XmlWriterSettings writ
         /// </summary>
         public XmlWriterSettings WriterSettings { get; }
 
+        protected virtual Type ResolveType(Type declaredType, Type runtimeType)
+        {
+            if (declaredType == null ||
+                declaredType == typeof(object))
+            {
+                if (runtimeType != null)
+                {
+                    return runtimeType;
+                }
+            }
+
+            return declaredType;
+        }
+
         /// <summary>
         /// Gets the type of the object to be serialized.
         /// </summary>
-        /// <param name="declaredType">The declared type.</param>
-        /// <param name="runtimeType">The runtime type.</param>
-        /// <returns>The type of the object to be serialized.</returns>
-        protected virtual Type GetSerializableType(Type declaredType, Type runtimeType)
+        /// <param name="type">The original type to be serialized</param>
+        /// <returns>The original or wrapped type provided by any <see cref="IWrapperProvider"/>s.</returns>
+        protected virtual Type GetSerializableType(Type type)
         {
-            Type type = declaredType;
-            if (declaredType == null || declaredType == typeof(object))
+            IWrapperProvider wrapperProvider = null;
+            foreach (var wrapperProviderFactory in WrapperProviderFactoryProvider.WrapperProviderFactories)
             {
-                if (runtimeType != null)
+                wrapperProvider = wrapperProviderFactory.GetProvider(new WrapperProviderContext(
+                                                                            type,
+                                                                            isSerialization:  true));
+                if (wrapperProvider != null)
                 {
-                    type = runtimeType;
+                    type = wrapperProvider.WrappingType;
+                    break;
```

Comment:
just return from inside here.

We don't follow the 'single exit point' rule, especially when it makes the code more complex.